### ISSUE TITLE
Some tests related to hash_map fail on big-endian platforms

### ISSUE THREAD

**Issue by musicinmybrain** [rkyv/rkyv#557]
Using a [`mock`](https://github.com/rpm-software-management/mock) chroot, emulating the big-endian `s390x` architecture via `qemu-user-static`:

```
# git clone https://github.com/rkyv/rkyv.git
# cd rkyv/
# cargo test
[…]

failures:

---- impls::std::collections::hash_map::tests::iter_seal stdout ----
thread 'impls::std::collections::hash_map::tests::iter_seal' panicked at rkyv/src/impls/std/collections/hash_map.rs:259:46:
called `Option::unwrap()` on a `None` value

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher_strings stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher_strings' panicked at rkyv/src/impls/std/collections/hash_map.rs:130:13:
assertion failed: b.contains_key(key.as_str())

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_string_string stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_string_string' panicked at rkyv/src/impls/std/collections/hash_map.rs:130:13:
assertion failed: b.contains_key(key.as_str())

---- impls::std::collections::hash_map::tests::get_seal stdout ----
thread 'impls::std::collections::hash_map::tests::get_seal' panicked at rkyv/src/impls/std/collections/hash_map.rs:239:62:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- impls::std::collections::hash_map::tests::large_hash_map stdout ----
thread 'impls::std::collections::hash_map::tests::large_hash_map' panicked at rkyv/src/impls/std/collections/hash_map.rs:130:13:
assertion failed: b.contains_key(key.as_str())

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_string_int stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_string_int' panicked at rkyv/src/impls/std/collections/hash_map.rs:130:13:
assertion failed: b.contains_key(key.as_str())

---- impls::std::collections::hash_map::tests::get_with stdout ----
thread 'impls::std::collections::hash_map::tests::get_with' panicked at rkyv/src/impls/std/collections/hash_map.rs:224:18:
called `Option::unwrap()` on a `None` value

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher' panicked at rkyv/src/api/test/inner_checked.rs:55:34:
assertion `left == right` failed
  left: {1: 2, 5: 6, 0: 1813857280, -128: 0}
 right: {3: 4, 7: 8, 1: 2, 5: 6}

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_zsts stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_zsts' panicked at rkyv/src/api/test/inner_checked.rs:55:34:
assertion `left == right` failed
  left: {(): 0}
 right: {(): 10}

---- impls::std::collections::hash_set::tests::roundtrip_hash_set stdout ----
thread 'impls::std::collections::hash_set::tests::roundtrip_hash_set' panicked at rkyv/src/impls/std/collections/hash_set.rs:110:17:
assertion failed: b.contains(key.as_str())

---- impls::std::with::tests::with_hash_map_mapkv stdout ----
thread 'impls::std::with::tests::with_hash_map_mapkv' panicked at rkyv/src/impls/std/with.rs:612:53:
called `Option::unwrap()` on a `None` value


failures:
    impls::std::collections::hash_map::tests::get_seal
    impls::std::collections::hash_map::tests::get_with
    impls::std::collections::hash_map::tests::iter_seal
    impls::std::collections::hash_map::tests::large_hash_map
    impls::std::collections::hash_map::tests::roundtrip_hash_map_string_int
    impls::std::collections::hash_map::tests::roundtrip_hash_map_string_string
    impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher
    impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher_strings
    impls::std::collections::hash_map::tests::roundtrip_hash_map_zsts
    impls::std::collections::hash_set::tests::roundtrip_hash_set
    impls::std::with::tests::with_hash_map_mapkv

test result: FAILED$<2>. 130 passed; 11 failed; 0 ignored; 0 measured; 0 filtered out; finished in 4.13s

error: test failed, to rerun pass `-p rkyv --lib`
```

These tests all pass on little-endian platforms.

**Comment by musicinmybrain** [comment]
```
# RUST_BACKTRACE=1 cargo test -p rkyv --lib
[…]

failures:

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher_strings stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher_strings' panicked at rkyv/src/impls/std/collections/hash_map.rs:130:13:
assertion failed: b.contains_key(key.as_str())
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: rkyv::impls::std::collections::hash_map::tests::assert_equal
             at ./src/impls/std/collections/hash_map.rs:130:13
   4: core::ops::function::Fn::call
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:79:5
   5: rkyv::api::test::inner_checked::roundtrip_with::{{closure}}
             at ./src/api/test/inner_checked.rs:39:9
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::api::test::inner_checked::roundtrip_with
             at ./src/api/test/inner_checked.rs:38:5
  10: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher_strings
             at ./src/impls/std/collections/hash_map.rs:200:9
  11: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher_strings::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:194:55
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  13: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_string_string stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_string_string' panicked at rkyv/src/impls/std/collections/hash_map.rs:130:13:
assertion failed: b.contains_key(key.as_str())
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: rkyv::impls::std::collections::hash_map::tests::assert_equal
             at ./src/impls/std/collections/hash_map.rs:130:13
   4: core::ops::function::Fn::call
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:79:5
   5: rkyv::api::test::inner_checked::roundtrip_with::{{closure}}
             at ./src/api/test/inner_checked.rs:39:9
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::api::test::inner_checked::roundtrip_with
             at ./src/api/test/inner_checked.rs:38:5
  10: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_string_string
             at ./src/impls/std/collections/hash_map.rs:163:9
  11: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_string_string::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:157:42
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  13: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_map::tests::iter_seal stdout ----
thread 'impls::std::collections::hash_map::tests::iter_seal' panicked at rkyv/src/impls/std/collections/hash_map.rs:259:46:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: core::option::unwrap_failed
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/option.rs:2020:5
   4: core::option::Option<T>::unwrap
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/option.rs:970:21
   5: rkyv::impls::std::collections::hash_map::tests::iter_seal::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:259:24
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::impls::std::collections::hash_map::tests::iter_seal
             at ./src/impls/std/collections/hash_map.rs:254:9
  10: rkyv::impls::std::collections::hash_map::tests::iter_seal::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:248:19
  11: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_map::tests::get_seal stdout ----
thread 'impls::std::collections::hash_map::tests::get_seal' panicked at rkyv/src/impls/std/collections/hash_map.rs:239:62:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: core::option::unwrap_failed
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/option.rs:2020:5
   4: core::option::Option<T>::unwrap
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/option.rs:970:21
   5: rkyv::impls::std::collections::hash_map::tests::get_seal::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:239:17
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::impls::std::collections::hash_map::tests::get_seal
             at ./src/impls/std/collections/hash_map.rs:237:9
  10: rkyv::impls::std::collections::hash_map::tests::get_seal::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:231:18
  11: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_map::tests::large_hash_map stdout ----
thread 'impls::std::collections::hash_map::tests::large_hash_map' panicked at rkyv/src/impls/std/collections/hash_map.rs:130:13:
assertion failed: b.contains_key(key.as_str())
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: rkyv::impls::std::collections::hash_map::tests::assert_equal
             at ./src/impls/std/collections/hash_map.rs:130:13
   4: core::ops::function::Fn::call
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:79:5
   5: rkyv::api::test::inner_checked::roundtrip_with::{{closure}}
             at ./src/api/test/inner_checked.rs:39:9
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::api::test::inner_checked::roundtrip_with
             at ./src/api/test/inner_checked.rs:38:5
  10: rkyv::impls::std::collections::hash_map::tests::large_hash_map
             at ./src/impls/std/collections/hash_map.rs:271:9
  11: rkyv::impls::std::collections::hash_map::tests::large_hash_map::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:266:24
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  13: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_string_int stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_string_int' panicked at rkyv/src/impls/std/collections/hash_map.rs:130:13:
assertion failed: b.contains_key(key.as_str())
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: rkyv::impls::std::collections::hash_map::tests::assert_equal
             at ./src/impls/std/collections/hash_map.rs:130:13
   4: core::ops::function::Fn::call
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:79:5
   5: rkyv::api::test::inner_checked::roundtrip_with::{{closure}}
             at ./src/api/test/inner_checked.rs:39:9
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::api::test::inner_checked::roundtrip_with
             at ./src/api/test/inner_checked.rs:38:5
  10: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_string_int
             at ./src/impls/std/collections/hash_map.rs:153:9
  11: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_string_int::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:146:39
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  13: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher' panicked at rkyv/src/api/test/inner_checked.rs:55:34:
assertion `left == right` failed
  left: {1: 2, 7: 8, 0: 0, 0: -1879048100}
 right: {5: 6, 3: 4, 7: 8, 1: 2}
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::assert_failed_inner
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:410:17
   3: core::panicking::assert_failed
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:365:5
   4: rkyv::api::test::inner_checked::roundtrip::{{closure}}
             at ./src/api/test/inner_checked.rs:55:34
   5: rkyv::api::test::inner_checked::roundtrip_with::{{closure}}
             at ./src/api/test/inner_checked.rs:39:9
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::api::test::inner_checked::roundtrip_with
             at ./src/api/test/inner_checked.rs:38:5
  10: rkyv::api::test::inner_checked::roundtrip
             at ./src/api/test/inner_checked.rs:55:5
  11: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher
             at ./src/impls/std/collections/hash_map.rs:190:9
  12: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:183:47
  13: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  14: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_map::tests::get_with stdout ----
thread 'impls::std::collections::hash_map::tests::get_with' panicked at rkyv/src/impls/std/collections/hash_map.rs:224:18:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: core::option::unwrap_failed
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/option.rs:2020:5
   4: core::option::Option<T>::unwrap
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/option.rs:970:21
   5: rkyv::impls::std::collections::hash_map::tests::get_with::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:220:28
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::impls::std::collections::hash_map::tests::get_with
             at ./src/impls/std/collections/hash_map.rs:219:9
  10: rkyv::impls::std::collections::hash_map::tests::get_with::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:204:18
  11: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_map::tests::roundtrip_hash_map_zsts stdout ----
thread 'impls::std::collections::hash_map::tests::roundtrip_hash_map_zsts' panicked at rkyv/src/api/test/inner_checked.rs:55:34:
assertion `left == right` failed
  left: {(): -8756894}
 right: {(): 10}
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::assert_failed_inner
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:410:17
   3: core::panicking::assert_failed
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:365:5
   4: rkyv::api::test::inner_checked::roundtrip::{{closure}}
             at ./src/api/test/inner_checked.rs:55:34
   5: rkyv::api::test::inner_checked::roundtrip_with::{{closure}}
             at ./src/api/test/inner_checked.rs:39:9
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::api::test::inner_checked::roundtrip_with
             at ./src/api/test/inner_checked.rs:38:5
  10: rkyv::api::test::inner_checked::roundtrip
             at ./src/api/test/inner_checked.rs:55:5
  11: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_zsts
             at ./src/impls/std/collections/hash_map.rs:170:9
  12: rkyv::impls::std::collections::hash_map::tests::roundtrip_hash_map_zsts::{{closure}}
             at ./src/impls/std/collections/hash_map.rs:167:33
  13: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  14: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::collections::hash_set::tests::roundtrip_hash_set stdout ----
thread 'impls::std::collections::hash_set::tests::roundtrip_hash_set' panicked at rkyv/src/impls/std/collections/hash_set.rs:110:17:
assertion failed: b.contains(key.as_str())
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: rkyv::impls::std::collections::hash_set::tests::roundtrip_hash_set::{{closure}}
             at ./src/impls/std/collections/hash_set.rs:110:17
   4: rkyv::api::test::inner_checked::roundtrip_with::{{closure}}
             at ./src/api/test/inner_checked.rs:39:9
   5: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   6: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   7: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   8: rkyv::api::test::inner_checked::roundtrip_with
             at ./src/api/test/inner_checked.rs:38:5
   9: rkyv::impls::std::collections::hash_set::tests::roundtrip_hash_set
             at ./src/impls/std/collections/hash_set.rs:106:9
  10: rkyv::impls::std::collections::hash_set::tests::roundtrip_hash_set::{{closure}}
             at ./src/impls/std/collections/hash_set.rs:98:28
  11: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- impls::std::with::tests::with_hash_map_mapkv stdout ----
thread 'impls::std::with::tests::with_hash_map_mapkv' panicked at rkyv/src/impls/std/with.rs:612:53:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: rust_begin_unwind
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:74:14
   2: core::panicking::panic
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/panicking.rs:148:5
   3: core::option::unwrap_failed
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/option.rs:2020:5
   4: core::option::Option<T>::unwrap
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/option.rs:970:21
   5: rkyv::impls::std::with::tests::with_hash_map_mapkv::{{closure}}
             at ./src/impls/std/with.rs:612:27
   6: rkyv::api::test::inner_checked::to_archived::{{closure}}
             at ./src/api/test/inner_checked.rs:25:9
   7: rkyv::api::test::outer_high::to_bytes
             at ./src/api/test/outer_high.rs:25:5
   8: rkyv::api::test::inner_checked::to_archived
             at ./src/api/test/inner_checked.rs:23:5
   9: rkyv::impls::std::with::tests::with_hash_map_mapkv
             at ./src/impls/std/with.rs:611:9
  10: rkyv::impls::std::with::tests::with_hash_map_mapkv::{{closure}}
             at ./src/impls/std/with.rs:598:29
  11: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
  12: core::ops::function::FnOnce::call_once
             at /builddir/build/BUILD/rust-1.81.0-build/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    impls::std::collections::hash_map::tests::get_seal
    impls::std::collections::hash_map::tests::get_with
    impls::std::collections::hash_map::tests::iter_seal
    impls::std::collections::hash_map::tests::large_hash_map
    impls::std::collections::hash_map::tests::roundtrip_hash_map_string_int
    impls::std::collections::hash_map::tests::roundtrip_hash_map_string_string
    impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher
    impls::std::collections::hash_map::tests::roundtrip_hash_map_with_custom_hasher_strings
    impls::std::collections::hash_map::tests::roundtrip_hash_map_zsts
    impls::std::collections::hash_set::tests::roundtrip_hash_set
    impls::std::with::tests::with_hash_map_mapkv
```

These fail on real `s390x` hardware too (as discovered while doing “scratch” builds of RPM packages for Fedora), but I don’t have interactive access to that. Reproducing in emulation is easier.

**Comment by djkoloski** [comment]
Fixed by 6d6a5461e42cac18707dfee42b9f41ee7d078866

**Comment by musicinmybrain** [comment]
I can confirm that 6d6a5461e42cac18707dfee42b9f41ee7d078866 fixes this issue in my environment. Thank you very much!
